<launch>

  <arg name="robot_model" default="$(find xacro)/xacro $(find turtlebot3_description)/urdf/turtlebot3_burger_gripper.urdf.xacro"/>

  <!-- Gazebo -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find multi_agent_avoid)/worlds/turtlebot3_stage_prime.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>  

  <!-- Robot 1 -->
  <arg name="r1"  default="robot1"/>
  <arg name="r1_x" default="1.0"/>
  <arg name="r1_y" default="0.0"/>
  <arg name="r1_z" default="0.0"/>
  <arg name="r1_yaw" default="0.0"/>

  <group ns = "$(arg r1)">
    <param name="robot_description" command="$(arg robot_model)"/>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg r1)" />
    </node>
    
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" 
	  args="-urdf -model $(arg r1) -x $(arg r1_x) -y $(arg r1_y) -z $(arg r1_z) -Y $(arg r1_yaw) -param robot_description" />
    
    <node name="multi_agent_cluster_node" pkg="multi_agent_cluster" type="multi_agent_cluster_node" output="screen">
      <param name="odometry_topic" type="string" value="/$(arg r1)/odom" />
      <param name="movement_topic" type="string" value="/$(arg r1)/cmd_vel" />
      <param name="top_scan_topic" type="string" value="/$(arg r1)/top_scan" />
      <param name="bottom_scan_topic" type="string" value="/$(arg r1)/bottom_scan" />
    </node>

  </group>

</launch>
